elasticdl/python/data/reader/odps_reader.py

```diff
@@ -59,6 +59,9 @@ def read_records(self, task):
             self._metadata.column_names = (
                 reader._odps_table.schema.names if columns is None else columns
             )
+            column_dtypes = []
+            for column_name in self._metadata.column_names:
```

Comment:
self._metadata.column_names = [reader._odps_table.schema[column_name] for column_name in self._metadata.column_names]